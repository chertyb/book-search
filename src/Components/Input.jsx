import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => {
  const { value, handleInputChange } = props;
  return <input type="text" value={value} onChange={handleInputChange} />;
};

Input.propTypes = {
  value: PropTypes.string,
  handleInputChange: PropTypes.func,
};

Input.defaultProps = {
  value: '',
  handleInputChange: () => {},
};

export default Input;
