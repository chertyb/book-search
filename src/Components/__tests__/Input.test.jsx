import React from 'react';
import ReactDom from 'react-dom';
import { shallow } from '../../enzyme';
import Input from '../Input';

describe('Input component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<Input />, div);
    expect(div).toMatchSnapshot();
    ReactDom.unmountComponentAtNode(div);
  });

  it('should display props value', () => {
    const value = 'Harry Potter';
    const wrapper = shallow(<Input value={value} />);
    expect(wrapper.prop('value')).toEqual(value);
  });
});
