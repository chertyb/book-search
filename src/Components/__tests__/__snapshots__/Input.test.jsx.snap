// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders without crashing 1`] = `
<div>
  <input
    type="text"
    value=""
  />
</div>
`;
